lib/dap/proto/ldap.rb

```diff
@@ -88,29 +159,73 @@ def self.parse_message(data)
         results['objectName'] = data.value[1].value[0].value
       end
 
-      attrib_hash = {}
+      if data.value[1].value[1]
+        attrib_hash = {}
 
-      # Handle PartialAttributeValues
-      data.value[1].value[1].each do |partial_attrib|
+        # Handle PartialAttributeValues
+        data.value[1].value[1].each do |partial_attrib|
 
-        value_array = []
-        attrib_type = partial_attrib.value[0].value
+          value_array = []
+          attrib_type = partial_attrib.value[0].value
 
-        partial_attrib.value[1].each do |part_attrib_value|
-          value_array.push(part_attrib_value.value)
+          partial_attrib.value[1].each do |part_attrib_value|
+            value_array.push(part_attrib_value.value)
+          end
+
+          attrib_hash[attrib_type] = value_array
         end
 
-        attrib_hash[attrib_type] = value_array
+        results['PartialAttributes'] = attrib_hash
       end
 
-      results['PartialAttributes'] = attrib_hash
-
-    elsif data.value[1].tag == 5
+    elsif data.value[1] && data.value[1].tag == 5
       # SearchResultDone found..
       result_type = 'SearchResultDone'
-      results['resultCode'] = data.value[1].value[0].value.to_i if data.value[1].value[0].value
-      results['resultMatchedDN'] = data.value[1].value[1].value if data.value[1].value[1].value
-      results['resultdiagMessage'] = data.value[1].value[2].value if data.value[1].value[2].value
+      ldap_result = data.value[1]
+
+      if ldap_result.value[0] && ldap_result.value[0].class == OpenSSL::ASN1::Sequence
+        # Encoding of the SearchResultDone seems to vary, this is RFC format
+        # of an LDAPResult ASN.1 structure in which the data is contained in a
+        # Sequence
+        results = parse_ldapresult(ldap_result.value[0])
+      elsif ldap_result.value[0]
+        # LDAPResult w/o outer Sequence wrapper, used by MS Windows
+        results = parse_ldapresult(ldap_result)
+      end
+      if data.value[2] && data.value[2].tag == 10
+        # Unknown structure for providing a response, looks like LDAPResult
+        # but placed at a higher level in the response, salvage what we can..
+        results['resultCode'] = data.value[2].value.to_i if data.value[2].value
+        results['resultDesc'] = RESULT_DESC[ results['resultCode'] ] if results['resultCode']
+        results['resultMatchedDN'] = data.value[3].value if data.value[3] && data.value[3].value
+        results['resultdiagMessage'] = data.value[4].value if data.value[4] && data.value[4].value
+      end
+
+    elsif data.value[1] && data.value[1].tag == 1
+      result_type = 'BindResponse'
+      results = parse_ldapresult(data.value[1])
+
+    elsif data.value[1] && data.value[1].tag == 2
+      result_type = 'UnbindRequest'
```

Comment:
No, in each case that I saw the payload was empty and no additional data provided.